lib/promscrape/targetstatus.go

```diff
@@ -19,9 +21,14 @@ var maxDroppedTargets = flag.Int("promscrape.maxDroppedTargets", 1000, "The maxi
 
 var tsmGlobal = newTargetStatusMap()
 
-// WriteHumanReadableTargetsStatus writes human-readable status for all the scrape targets to w.
-func WriteHumanReadableTargetsStatus(w io.Writer, showOriginalLabels bool) {
-	tsmGlobal.WriteHumanReadable(w, showOriginalLabels)
+// WriteHumanReadableTargetsStatus writes human-readable status for all the scrape targets to w with given format and options.
+func WriteHumanReadableTargetsStatus(w io.Writer, showOriginalLabels, showOnlyUnhealthy bool, format string) {
+	switch format {
+	case "plain":
+		tsmGlobal.WriteTargetsPlain(w, showOriginalLabels)
```

Comment:
It is better from maintainability PoV setting `Content-Type` headers here instead of setting them at call sites for `WriteHumanReadableTargetsStatus`. Just use `w http.ResponseWriter` arg for this function.

Reply:
done